### PR TITLE
allow custom override for entity star icon

### LINKED ISSUE #26385: 🐛 Bug Report: Favourite icon components not using theme colors
### 📜 Description

The colour of the `Star` component in catalog, home and techdocs are using a hard-coded yellow colour instead of the primary theme colour.

### 👍 Expected behavior

The various `Star` components should use overrideable theme colours to allow for customisation.

For example (see favourite icon - `Star`):
image

### 👎 Actual Behavior with Screenshots

Observe the yellow star in the catalog entity table:
image

### 👟 Reproduction steps

Visit any `/catalog` or `/docs` to observe the color of the favourite icon (`Star`).

You can find all occurrences by searching for `#f3ba37` throughout the code base.

### 📃 Provide the context for the Bug.

The current hard-coded yellow color conflicts with our "warning" theme color.  

### 🖥️ Your Environment

N/A

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by stephenglass**
## Hey, I just made a Pull Request!

Changes in this PR:

- Adds `star` and `unstarred` overridable icons. Default icon is the existing star.
- Updates existing usage of stars to use the new Backstage `star` and `unstarred` icons and theme color

There is no visual change when using default (non-themed) Backstage.

Example overriding default star icon to a "favorite" (heart) icon:

App.tsx
```ts
import FavoriteIcon from '@material-ui/icons/Favorite';
import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';

const app = createApp({
  apis,
  plugins: Object.values(plugins),
  icons: {
    // Custom icon example
    star: FavoriteIcon,
    unstarred: FavoriteBorderIcon,
  },
  ...
});
```

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/theme/src/base/types.ts [hunk 1]
```diff
@@ -82,6 +82,9 @@ export type BackstagePaletteAdditions = {
     closeButtonColor?: string;
     warning?: string;
   };
+  entityStarButton: {
```
Comment (jhaals) [inline packages/theme/src/base/types.ts:85]:
```suggestion
  starButton: {
```
I wonder if it's worth dropping `entity` from the name and just naming it starButton? I'm thinking that there might be cases where you want to use the same styling but for some other type of star that does not strictly relate to an entity.

**Review by jhaals — COMMENTED** [review]
Nice work with this!

**Comment by benjdlambert** [comment]
@stephenglass sorry I think we should have been more clear with the other PR. @callebstrom is gonna create another PR to fix it a different way after we chatted offline for a bit.

It's gonna be similar to this, but I think we're gonna opt for not storing this as a value in the `palette`. There's already some accent colours we could use in the `palette`, but unsure of the implications of doing that.

Gonna use something similar to [these docs](https://backstage.io/docs/getting-started/app-custom-theme/#overriding-backstage-and-material-ui-components-styles) in order to make the colour configurable at the theme level instead.

It's up to you if you wanna keep this open or not, maybe we could close this in favour of @callebstrom's when it lands?

**Comment by callebstrom** [comment]
@benjdlambert There might still be value in keeping the `AppIcons` aspect of this PR actually, as I will only be tackling the color aspects using `makeStyles` and a custom component. I could utilise the app icons in my custom component to also allow overriding of the underlying icon, but still control the behaviour favorite marking. Wdyt?

**Comment by benjdlambert** [comment]
@callebstrom yeah, probably makes sense to allow the icon component itself to be overridden too, let's merge your PR first and then @stephenglass if you want to rebase on top and just pick out the `AppIcon` changes on top? :pray:

**Comment by stephenglass** [comment]
Sounds good! 👍

**Comment by callebstrom** [comment]
For transparency: here is the PR for the `FavoriteToggle` component https://github.com/backstage/backstage/pull/26484

**Comment by stephenglass** [comment]
PR has been updated to include only the custom icon overrides. Hope it's all good!

**Comment by mounicarajput** [comment]
Great work with this!

**Review by benjdlambert — APPROVED** [review]
Nice! thanks @stephenglass! :tada:

**Pull request merged**